src/indexer/IndexerStreamProcessor.ts

```diff
@@ -0,0 +1,60 @@
+import { EventsStore } from "../events.store";
+import { Poll } from "../poll";
+import { IndexerStorage } from "./indexerSorage";
+import { pollForEvents } from '../poll-for-events';
+import { IndexDefinition } from "../models/IndexDefinition";
+import { parseGetParams } from "../parse-http-params";
+import { DefinedHttpError } from "restify-errors";
+import bunyan from "bunyan";
+import { logger } from '../logger';
+import lodash from 'lodash';
+import async from "async";
+
+
+export class IndexerStreamProcessor {
+
+  poll: Poll;
+  store: EventsStore;
+  indexerStorage: IndexerStorage;
+  log: bunyan;
+
+
+  constructor(poll: Poll, store: EventsStore, indexerStorage: IndexerStorage, log: bunyan = logger) {
+    this.poll = poll;
+    this.store = store;
+    this.indexerStorage = indexerStorage;
+    this.log = log;
+  }
+
+  //process last fetched events, and add them to storageIndexer
+  processEvents(indexDefinition: IndexDefinition, cb: (e: Error | null, results: any | null) => void) {
+    let params = parseGetParams({ channel: indexDefinition.channel, clientId: indexDefinition.id });
```

Comment:
Here, I needed to provide the same parameters object to the pollEvents params .. since i don't have the 'after, limit, afterExplicitlySet) and I didn't want to re-write the code for setting the default values for those. So I used the function to return that object.